src/main/java/org/snomed/snowstorm/core/data/services/classification/RemoteClassificationServiceClient.java

```diff
@@ -1,48 +1,75 @@
+import org.apache.activemq.command.ActiveMQTextMessage;
 import org.apache.tomcat.util.http.fileupload.util.Streams;
 import org.elasticsearch.common.Strings;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.snomed.snowstorm.core.data.services.classification.pojo.ClassificationStatusResponse;
-import org.snomed.snowstorm.core.util.TimerUtil;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.boot.web.client.RestTemplateBuilder;
 import org.springframework.core.io.FileSystemResource;
 import org.springframework.http.*;
+import org.springframework.jms.annotation.JmsListener;
+import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Service;
 import org.springframework.util.LinkedMultiValueMap;
 import org.springframework.util.MultiValueMap;
 import org.springframework.web.client.ResponseExtractor;
 import org.springframework.web.client.RestClientException;
 import org.springframework.web.client.RestTemplate;
 
+import javax.jms.JMSException;
 import java.io.*;
 import java.nio.file.Files;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
 
 @Service
 class RemoteClassificationServiceClient {
 
 	private final RestTemplate restTemplate;
-
 	private final Logger logger = LoggerFactory.getLogger(getClass());
+	private final ObjectMapper objectMapper = new ObjectMapper();
+	private final Map<String, ClassificationStatusResponse> classificationStatuses = new HashMap<>(); // ClassificationStatusResponse.id  => ClassificationStatusResponse
+	private final String responseMessageQueue;
 
 	private static final HttpHeaders MULTIPART_HEADERS = new HttpHeaders();
 	static {
 		MULTIPART_HEADERS.setContentType(MediaType.MULTIPART_FORM_DATA);
 	}
 
 	public RemoteClassificationServiceClient(@Value("${classification-service.url}") String serviceUrl,
-			@Value("${classification-service.username}") String serviceUsername,
-			@Value("${classification-service.password}") String servicePassword) {
-
+											 @Value("${classification-service.username}") String serviceUsername,
+											 @Value("${classification-service.password}") String servicePassword,
+											 @Value("${classification-service.job.destination-response}") String responseMessageQueue) {
+		this.responseMessageQueue = responseMessageQueue;
 		restTemplate = new RestTemplateBuilder()
 				.rootUri(serviceUrl)
 				.basicAuthentication(serviceUsername, servicePassword)
 				.build();
 	}
 
+	@JmsListener(destination = "${classification-service.job.destination-response}")
+	void messageConsumer(ActiveMQTextMessage activeMQTextMessage) throws JMSException, JsonProcessingException {
+		try {
+			ClassificationStatusResponse response = objectMapper.readValue(activeMQTextMessage.getText(), ClassificationStatusResponse.class);
+			classificationStatuses.put(response.getId(), response);
+		} catch (JsonParseException | JsonMappingException e) {
+			// ignore
+		}
+	}
```

Comment:
Consume a message from the new status queue, wrapped in a try-catch in case anything ends up there that shouldn't have.